src/metabase/task/index_values.clj

```diff
@@ -0,0 +1,116 @@
+(ns metabase.task.index-values
+  (:require
+   [clojurewerkz.quartzite.conversion :as qc]
+   [clojurewerkz.quartzite.jobs :as jobs]
+   [clojurewerkz.quartzite.schedule.cron :as cron]
+   [clojurewerkz.quartzite.triggers :as triggers]
+   [metabase.driver :as driver]
+   [metabase.models.card :refer [Card]]
+   [metabase.models.model-index :as model-index :refer [ModelIndex]]
+   [metabase.query-processor.timezone :as qp.timezone]
+   [metabase.task :as task]
+   [metabase.util :as u]
+   [metabase.util.i18n :refer [trs]]
+   [metabase.util.log :as log]
+   [toucan2.core :as t2])
+  (:import
+   (java.util TimeZone)
+   (org.quartz ObjectAlreadyExistsException)))
+
+(set! *warn-on-reflection* true)
+
+#_{:clj-kondo/ignore [:unused-private-var]}
+(def ^:private refreshable-states
+  "States of a model index that are refreshable."
+  #{"indexed" "initial" "error" "overflow"})
+
+(defn should-deindex?
+  "Whether to unindex the the model indexing job.
+  If the model does not exist, the model-index record, or if the model has reverted to a question."
```

Comment:
I don't understand this sentence ("the model-index record") and it doesn't say anything about the model being archived.